### PR TITLE
Rails webpacker (take two)

### PR DESCRIPTION
**Pull request by rsanheim**
**Story card:** [ch4400](https://app.clubhouse.io/simpledotorg/story/4400/create-new-v2-layout-that-uses-a-webpack-bundle)

### This will require a Node 12 upgrade on our servers: https://github.com/simpledotorg/deployment/pull/358

Trying to get webpack into the app via [webpacker](https://github.com/rails/webpacker).  The goal is to have webpack serve up a bundle via a new 'v2' layout, and to leave the legacy Sprockets asset pipeline in place for all old pages as-is.

@claudiovallejo and I hit an issue where we aren't able to import bootstrap into the new CSS bundle...I'm guessing some configuration is missing here to wire up CSS / SCSS loading into webpack.  We haven't yet found the right tutorial to follow or the right configuration we need to add.

### How to try this in dev

```
yarn install
brew install sass/sass/sass
bin/webpack --watch # run this in a terminal you leave up and running...
rails server
```

### Some useful guides

* https://rossta.net/blog/webpacker-with-bootstrap.html
* https://rossta.net/webpack-on-rails/

### CODE REVIEW

**Comment by rsanheim** [comment]
Blocked by https://github.com/simpledotorg/deployment/pull/358

File: app/views/api/v3/help/show.html.erb [hunk 1]
```diff
@@ -53,7 +53,7 @@
           <p>
             <%= t("help.support.contents_html") %>
           </p>
-          <a href="<%= ENV['HELP_SCREEN_SUPPORT_GROUP_URL'] %>" class="support-button <%= CountryConfig.current[:abbreviation] %>-button">
+        <a href="<%= ENV['HELP_SCREEN_SUPPORT_GROUP_URL'] %>" class="support-button <%= CountryConfig.current[:abbreviation] %>-button">
```
Comment (harimohanraj89) [inline app/views/api/v3/help/show.html.erb:56]: Unintentional diff?

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<h1>Home#index</h1>
+
+<p>Welcome to new home page 😎</p>
+
+<strong><a href="https://zombo.com">You can do anything at Zombo.com.</a>
```
Comment (harimohanraj89) [inline app/views/home/index.html.erb:5]: Yassss.

File: package.json [hunk 3]
```diff
@@ -2,7 +2,28 @@
   "name": "simple-server",
   "private": true,
   "dependencies": {
+    "@rails/webpacker": "^6.0.0-beta.7",
+    "bootstrap": "^5.0.2",
+    "css-loader": "^6.2.0",
+    "css-minimizer-webpack-plugin": "^3.0.2",
+    "mini-css-extract-plugin": "^2.1.0",
     "numeral": "^2.0.6",
-    "tablesort": "^5.2.0"
```
Comment (harimohanraj89) [inline package.json:6]: Is tablesort still accessible via... sprockets(?) on existing pages?

File: Gemfile [hunk 4]
```diff
@@ -70,22 +74,19 @@ gem "scientist"
...
 gem "uuidtools", require: false
 gem "view_component", require: "view_component/engine"
+gem "webpacker", "~> 6.x"
 gem "whenever", require: false
 gem "wkhtmltoimage-binary"
-gem "memery"
-gem "bootstrap-select-rails"
-gem "render_async"
-gem "rack-attack"
-gem "squid"
```
Comment (harimohanraj89) [inline Gemfile:88]: Ha, thank you for alphabetizing this.

File: app/javascript/packs/application.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+Any CSS added to this file or imported from this file, e.g. `@import '../stylesheets/my-css.css'`,
+will be included in the "application" pack. Any CSS imported from application.js or as part of the
+application.js dependency graph, e.g. `import '../stylesheets/my-css.css'` will also be included
+in the "application" pack.
+
+To reference this file, add <%= stylesheet_pack_tag 'application' %> to the appropriate
+layout file, like app/views/layouts/application.html.erb
+*/
+
+@import 'bootstrap/scss/bootstrap';
```
Comment (harimohanraj89) [inline app/javascript/packs/application.scss:11]: Let's use some var or mixin from bootstrap for good measure? Make sure the wires are all hooked up correctly?

**Review by harimohanraj89 — COMMENTED** [review]
Looks like fairly standard webpack stuff. Is the bootstrap inclusion working yet?

File: package.json (hunk 3 above)
Comment (rsanheim) [reply package.json:6]: Hrm, good catch just saw this after merge...I'll take a look.

Comment (rsanheim) [reply package.json:6]: Just verified, it still works.

I don't really know how all our old assets work TBH, I don't think they really use package.json at all for them.  Will be nice to remove sprockets altogether.

**Pull request merged**